woocommerce/woocommerce-ios
# [Mobile Payments] Improved minimum WCPay plugin version comparison

Fixes #5357 

Changes:
- Adds a new String extension to do version comparison
- Changes onboarding to use that instead
- Expands onboarding unit tests to cover major.minor and major.minor.patch versions better

To test:
- Ensure unit tests pass
- Edit the WCPay plugin to change the version in the header to 3.1.8
- Launch the app, enter settings and in-person payments and confirm "unsupported ... version" is displayed
- Edit the WCPay plugin to change the version in the header to 3.2
- Refresh the "unsupported ... version" screen and ensure it still says "unsupported ... version"
- Edit the WCPay plugin to change the version in the header to 3.2.1 (the minimum currently required)
- Refresh the "unsupported ... version" screen and ensure it shows manage/buy/manual
- Edit the WCPay plugin to change the version in the header to 3.3
- Restart the app, enter settings and in-person payments and confirm it shows manage/buy/manual
- For bonus points, edit `minimumSupportedWCPayVersion` in `CardPresentPaymentsOnboardingUseCase` to 3.3, restart the app, enter settings and in-person payments and confirm it shows manage/buy/manual

Update release notes:
